src/kixi/hecuba/onyx/jobs/weather_connector.clj

```diff
@@ -0,0 +1,129 @@
+                      {:form-params {:Type "Observation"
+                                     :PredictionSiteID siteid
+                                     :ObservationSiteID siteid
+                                     :Date querydate ;; dd/mm/yyyy
+                                     :PredictionTime querytime ;; 0000
+                                     }
+                       :follow-redirects true})))
+
+(defn get-met-office-csv-for-day-hour
+  "Get Metoffice data as a string"
+  [querydate querytime siteid]
+  (timbre/info "get-met-office-csv-for-day-hour")
+  (->> (met-office-post-request querydate querytime siteid)
+       (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
+       (client/get)
+       :body))
+
+
+(defn pull-weather-station-day-data [querydate siteid]
+  (timbre/info "pull-weather-station-day-data")
+  (into [] (keep (fn [hour] (try (-> (get-met-office-csv-for-day-hour querydate (format "%02d00" hour) siteid)
+                                     (convert-day-data-to-seq)
+                                     (extract-temp-date-time-from-seq))
+                                 (catch Exception e (str "Exception caught: " (.getMessage e)))))
+                 (range 0 24))))
+
+(defn create-degree-day-measurement [measurements]
+  (timbre/info "create-degree-day-measurement")
+  (let [min-max (get-max-and-min measurements)]
+    {:value (calc-degreedays-mckiver (:min min-max) (:max min-max))
+     :type "Temperature_degreedays"
+     :timestamp (:timestamp (first measurements))}))
+
+(defn create-measurements [measurement-data]
+  (timbre/info "create-measurements")
+  (map (fn [reading]
+         (let [{:keys [temperature date time]} (first reading)]
+           {:value temperature
+            :type "Temperature"
+            :timestamp (f/unparse (f/formatters :date-time)
+                                  (f/parse tformat (str date " " time)))})) measurement-data))
+
+(defn build-payload [weather-date fn-data]
+  (timbre/info "build-payload")
+  (let [measurements (create-measurements (pull-weather-station-day-data weather-date (:property-code fn-data)))
+        degree-day (create-degree-day-measurement measurements)]
+    {:kafka-payload fn-data
+     :entity-id fn-data
+     :measurements measurements
+     :degree-day degree-day}))
+
+;; use {:done true} to tell outgoing workflow the Kafka message is acknowledged and dealt with.
+(defn get-data [fn-data]
+  (let [payload (build-payload
+                 (f/unparse dformat (t/minus (t/now) (t/days 2)))
+                 fn-data)]
+    {:message payload
+     :entity fn-data
+     :done true}))
+
+
+(s/defn run-weather
+  ([task-name :- s/Keyword task-opts]
+   {:task {:task-map (merge {:onyx/name task-name
+                             :onyx/type :function
+                             :onyx/fn :kixi.hecuba.onyx.jobs.weather-connector/get-data}
+                            task-opts)}}))
```

Comment:
I disagree. You need all 24 measurements, splitting up into further tasks will make degree day calculation harder to do. So the function does all 24 calls, filters the data it needs and calculates the degree day. I'd prefer to leave this alone, it's working well under testing.